CHANGELOG.md

```diff
@@ -38,6 +38,7 @@ For older changes see the [archived Singularity change log](https://github.com/a
 
 ### Other changes
 
+- Fix `GOCACHE` settings for golang build on PPA build environment.
```

Comment:
Since this is going into the release-1.1 branch, please instead add a new `### Bug fixes` section below `### Other changes` and put your changelog entry there.

Reply:
Thank you @DrDaveD , I'll add bug fixes section and move may entry there.

This PR fixes golang build problem on PPA environment by properly setting GOCACHE. For example, if we raised minimum golang version to 1.19.x, Ubuntu 22.04 has 1.18.1, we need to build golang on Ubuntu 22.04 as well. This PR also helps that situation as well.

#1106 I observed and found out this problem on Ubuntu 20.04, but this also happens on Ubuntu 22.04 when we raised minimum golang version newer than system golang.